client-spark/common/src/main/java/org/apache/spark/shuffle/writer/DataPusher.java

```diff
@@ -88,14 +88,25 @@ public CompletableFuture<Long> send(AddBlockEvent event) {
         () -> {
           String taskId = event.getTaskId();
           List<ShuffleBlockInfo> shuffleBlockInfoList = event.getShuffleDataInfoList();
+          SendShuffleDataResult result = null;
           try {
-            SendShuffleDataResult result =
+            result =
                 shuffleWriteClient.sendShuffleData(
                     rssAppId, shuffleBlockInfoList, () -> !isValidTask(taskId));
             putBlockId(taskToSuccessBlockIds, taskId, result.getSuccessBlockIds());
             putFailedBlockSendTracker(
                 taskToFailedBlockSendTracker, taskId, result.getFailedBlockSendTracker());
           } finally {
+            Set<Long> succeedBlockIds = result.getSuccessBlockIds();
+            if (succeedBlockIds != null && !succeedBlockIds.isEmpty()) {
```

Comment:
CollectionUtils has a similar method to simplify code.

Reply:
Got it

Reply:
Done